src/modules/jcms_notifications/tests/src/Unit/ArticleCrudServiceTest.php

```diff
@@ -0,0 +1,153 @@
+
+use Drupal\Core\DependencyInjection\ContainerBuilder;
+use Drupal\Tests\UnitTestCase;
+use Drupal\jcms_notifications\ArticleCrudService;
+use \Mockery as m;
+
+/**
+ * Class ArticleCrudServiceTest
+ *
+ * @package Drupal\Tests\jcms_notifications\Unit
+ */
+class ArticleCrudServiceTest extends UnitTestCase {
+
+  /**
+   * @var \Drupal\jcms_notifications\ArticleCrudService
+   *   The class we're testing.
+   */
+  protected $articleCrudService;
+
+  /**
+   * @var \Drupal\jcms_notifications\FetchArticleService
+   */
+  protected $fetchArticleService;
+
+  /**
+   * @var \Drupal\Core\Entity\EntityTypeManager
+   */
+  protected $entityTypeManager;
+
+  /**
+   * @var \Drupal\Core\Entity\Query\QueryFactory
+   */
+  protected $entityQuery;
+
+  public function setUp() {
+    $this->fetchArticleService = m::mock('\Drupal\jcms_notifications\FetchArticleService');
+    $this->entityTypeManager = m::mock('\Drupal\Core\Entity\EntityTypeManager');
+    $this->entityQuery = m::mock('\Drupal\Core\Entity\Query\QueryFactory');
+    $container = new ContainerBuilder();
+    $container->set('entity.query', $this->entityQuery);
+    $container->set('entity.manager', $this->entityTypeManager);
+    \Drupal::setContainer($container);
+    $this->articleCrudService = new ArticleCrudService($this->fetchArticleService, $this->entityTypeManager);
+  }
+
+  public function tearDown() {
+    m::close();
+  }
+
+  /**
+   * @test
+   * @covers \Drupal\jcms_notifications\ArticleCrudService::__construct()
+   */
+  public function testCrudArticleService() {
+    $actual = new ArticleCrudService($this->fetchArticleService, $this->entityTypeManager);
+    $this->assertInstanceOf('\Drupal\jcms_notifications\ArticleCrudService', $actual);
+  }
+
+  /**
+   * @test
+   * @covers \Drupal\jcms_notifications\ArticleCrudService::crudArticle()
+   * @covers \Drupal\jcms_notifications\ArticleCrudService::nodeExists()
+   * @covers \Drupal\jcms_notifications\ArticleCrudService::deleteArticle()
+   */
+  public function testCrudArticleDelete() {
+    $this->entityQuery->shouldReceive('get')->once()->andReturn($this->entityQuery);
+    $this->entityQuery->shouldReceive('condition')->once()->andReturn($this->entityQuery);
+    $this->entityQuery->shouldReceive('execute')->once()->andReturn([1234]);
+    $this->entityTypeManager->shouldReceive('getStorage')->once()->andReturn($this->entityTypeManager);
+    $this->entityTypeManager->shouldReceive('delete')->once()->andReturn(NULL);
+    $acutal = $this->articleCrudService->crudArticle([
+      'id' => '123abc',
+      'action' => 'del',
+    ]);
+    $this->assertNull($acutal);
+  }
+
+  /**
+   * @test
+   * @covers \Drupal\jcms_notifications\ArticleCrudService::crudArticle()
+   * @covers \Drupal\jcms_notifications\ArticleCrudService::nodeExists()
+   * @covers \Drupal\jcms_notifications\ArticleCrudService::updateArticle()
+   */
+  public function testCrudArticleUpdate() {
+    $node = m::mock('\Drupal\node\Entity\Node');
```

Comment:
recommending to drop this as a unit tests (mocking a dozen of objects to make it work) and implement it as an integration test inside this project instead (using the real MySQL database underneath by driving method calls to an ArticleCrudService object instead, everything in the single PHPUnit process)